Source/CombatExtended/CombatExtended/ArmorUtilityCE.cs

```diff
@@ -101,30 +101,69 @@ public static DamageInfo GetAfterArmorDamage(DamageInfo originalDinfo, Pawn pawn
                     {
-                        shieldAbsorbed = true;
-                        armorDeflected = true;
-                        dinfo.SetAmount(0);
+                        //Deflected by sharp armor, check for blunt armor
+                        if (dinfo.Def.armorCategory.armorRatingStat == StatDefOf.ArmorRating_Sharp)
+                        {
+                            if (deflectionComp != null)
+                            {
+                                deflectionComp.deflectedSharp = true;
+                            }
+                        }
+                        // Hit was deflected, convert damage type
+                        dinfo = GetDeflectDamageInfo(dinfo, hitPart, ref dmgAmount, ref penAmount);
+                        //Check if converted blunt damage could penetrate the shield
+                        if (!TryPenetrateArmor(dinfo.Def, shield.GetStatValue(dinfo.Def.armorCategory.armorRatingStat), ref penAmount, ref dmgAmount))
+                        {
+                            shieldAbsorbed = true;
+                            armorDeflected = true;
+                            dinfo.SetAmount(0);
+                            // Apply secondary damage to shield
+                            if (dinfo.Weapon?.projectile is ProjectilePropertiesCE props && !props.secondaryDamage.NullOrEmpty())
+                            {
+                                foreach (var sec in props.secondaryDamage)
+                                {
+                                    if (shield.Destroyed || !Rand.Chance(sec.chance))
+                                    {
+                                        break;
+                                    }
+                                    var secDinfo = sec.GetDinfo();
+                                    var pen = secDinfo.ArmorPenetrationInt; //GetPenetrationValue(originalDinfo);
+                                    var dmg = (float)secDinfo.Amount;
+                                    TryPenetrateArmor(secDinfo.Def, shield.GetStatValue(secDinfo.Def.armorCategory.armorRatingStat), ref pen, ref dmg, shield);
+                                }
+                            }
 
-                        // Apply secondary damage to shield
-                        if (dinfo.Weapon?.projectile is ProjectilePropertiesCE props && !props.secondaryDamage.NullOrEmpty())
+                            return dinfo;
+                        }
+                        //Blunt damage penetrated the shield, apply the damage to right arm
+                        else
                         {
-                            foreach (var sec in props.secondaryDamage)
+                            shieldAbsorbed = true;
+                            //I'm not sure whether pawn having no arms is able to use the shield
```

Comment:
I believe the check for being able to wield a shield checks for `LeftShoulder`; may be wrong, will check.

Reply:
Could add a check if pawn is only equipping shield without one-handed weapon. Could add some blunt armor as a simulation to extra impact resistance